### PR TITLE
added authentication  and tested POST, GET requests in the React app

### PR DESCRIPTION
**Pull request by singhrohan62**
Tested authentication of the user and POST, GET requests in the React application.

## Description
Added small changes to the `Login` component, tested API calls by making POST and GET requests from the application with the help of axios. POST request was regarding getting logged in and GET request was made to test GET request with token in request header.

I didn't created an issue for this PR, since it is a PR with small changes to the application.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: components/frontend-react/src/containers/LoginComponent.js [hunk 1]
```diff
@@ -1,19 +1,52 @@
 import React from 'react';
 import Grid from '@material-ui/core/Grid';
+import axios from 'axios';
 
 import Appbar from '../components/Appbar';
 import BassaIntroBox from '../components/BassaIntroBox';
 import UserSignup from '../components/UserSignup';
 
+const APIURL = "http://0.0.0.0:5000/api";
```
Comment (iamchathu) [inline components/frontend-react/src/containers/LoginComponent.js]: Get the URL from .env.

Create .env file and have REACT_APP_API_URL or some env variable in the file. Create React app will autoload .env file and REACT_APP_* variables when we run and build application.

further reading https://facebook.github.io/create-react-app/docs/adding-custom-environment-variables

Comment (singhrohan62) [reply components/frontend-react/src/containers/LoginComponent.js]: Thanks @iamchathu for your guidance and advice. I will add .env file for that.

File: components/frontend-react/src/components/UserSignup.js [hunk 2]
```diff
@@ -43,8 +43,8 @@ const useStyles = makeStyles(theme => ({
 const UserSignup = (props) => {
 
   const handleSubmitButton = (event) => {
-    event.preventDefault()
-    props.onClickSubmit()
+    event.preventDefault();
```
Comment (JaDogg) [inline components/frontend-react/src/components/UserSignup.js]: ✅

File: components/frontend-react/src/containers/LoginComponent.js [hunk 3]
```diff
@@ -1,19 +1,49 @@
...
+        headers: {'Content-Type': 'multipart/form-data' }
+      })
+    .then(res => {
+      makeGetRequest(res.headers.token);
+    })
+    .then(err => console.log(err));
+  }
+
+  const makeGetRequest = (token) => {
+    //Testing a GET request
```
Comment (JaDogg) [inline components/frontend-react/src/containers/LoginComponent.js]: What is the purpose of this comment? If it is not necessary remove it?

Comment (singhrohan62) [reply components/frontend-react/src/containers/LoginComponent.js]: @JaDogg I will remove this function in the next PR. The function was created just to test whether GET requests are being made successfully.

**Comment by agentmilindu** [comment]
I think it is a good practice to add the `.env` file to `.gitignore` file to avoid someone mistakenly committing their `.env` file

**Comment by agentmilindu** [comment]
Oh, it seems it is already there in the `.gitignore` https://github.com/scorelab/Bassa/blob/develop/.gitignore#L74

**Comment by agentmilindu** [comment]
@JaDogg @charithccmc Can you approve the PR?

**Pull request merged**